core/node/eth_sender/src/aggregator.rs

```diff
@@ -199,11 +241,68 @@ impl Aggregator {
             ready_for_execute_batches,
             last_sealed_l1_batch,
         )
-        .await;
+        .await?;
+
+        let start_index = self.config.priority_tree_start_index;
+        let priority_merkle_tree = self.get_or_init_tree().await;
+
+        let mut priority_ops_proofs = vec![];
+        for batch in l1_batches.iter() {
+            let priority_tree_start_index = if let Some(start_index) = start_index {
+                start_index
+            } else {
+                // Start index not present, we push empty proof.
+                priority_ops_proofs.push(Default::default());
+                continue;
```

Comment:
Not saying this is necessarily incorrect, but looks sus (if I read the code correctly): Since `start_index` isn't changed in the loop, why can't it be refactored out of it, so that the loop isn't executed at all if `start_index == None`, and instead `priority_ops_proofs` is filled with default data? If this is intended, IMO, such refactoring would make the intent more explicit.